doc/source/serve/architecture.md

```diff
@@ -49,7 +49,7 @@ Each replica maintains a queue of requests and executes requests one at a time,
 using `asyncio` to process them concurrently. If the handler (the deployment function or the `__call__` method of the deployment class) is declared with `async def`, the replica will not wait for the
 handler to run.  Otherwise, the replica will block until the handler returns.
 
-When making a request via [ServeHandle](serve-handle-explainer) instead of HTTP, the request is placed on a queue in the ServeHandle, and we skip to step 3 above.
+When making a request via a [DeploymentHandle](deployment-handle-explainer) instead of HTTP for [model composition](serve-model-composition), the request is placed on a queue in the `DeploymentHandle`, and we skip to step 3 above.
```

Comment:
I think (deployment-handle-explainer) is still pointing to the section in model composition